app/scripts/controllers/buildConfig.js

```diff
@@ -74,6 +109,7 @@ angular.module('openshiftConsole')
                 };
               }
               $scope.buildConfig = buildConfig;
+              $scope.updatedBuildConfig = angular.copy(buildConfig);
```

Comment:
if anything updates the build config it could wipe out any changes you've made in the form, background controllers could potentially trigger an update to the BC's status (example, the latest build number that was run)